src/config/enterprise.rs

```diff
@@ -103,6 +103,47 @@ impl Default for Options {
     }
 }
 
+impl Options {
+    pub fn merge(&mut self, other: Self) {
+        self.enabled = other.enabled;
+        self.enable_logs_reporting = other.enable_logs_reporting;
+        if other.site.is_some() {
```

Comment:
I feel like we should check the contents of `site`, `region`, `endpoint` and error (or at least leave a `warn!` log) if they're mismatched when trying to merge. If a user set specific options for these across configurations, it seems accidental for them to start Vector with both (or more) disparate options.